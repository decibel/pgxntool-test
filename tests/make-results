#!/bin/bash

trap 'echo "ERROR: $BASH_SOURCE: line $LINENO" >&2' ERR
set -o errexit -o errtrace -o pipefail
#set -o xtrace -o verbose

if [ "$1" == "-v" ]; then
  verboseout=1
  shift
fi
BASEDIR=`cd ${0%/*}; pwd`

. $BASEDIR/../.env
. $TOPDIR/lib.sh

cd $TEST_REPO

out Mess with output to test make results
echo >> test/expected/pgxntool-test.out

out Test make results
make test
out -v ^^^ Should have a diff ^^^
make results
make test
out -v ^^^ Should be clean output, BUT NOTE THERE WILL BE A FAILURE DIRECTLY ABOVE! ^^^

check_log

# vi: expandtab sw=2 ts=2
