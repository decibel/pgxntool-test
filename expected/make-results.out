# Mess with output to test make results
# Test make results
rm -rf  sql/pgxntool-test--0.1.0.sql
rm -rf results/ regression.diffs regression.out tmp_check/ log/
cp sql/pgxntool-test.sql sql/pgxntool-test--0.1.0.sql
@INSTALL@ -c -d '@PG_LOCATION@/share/extension'
@INSTALL@ -c -m 644 ./sql/pgxntool-test--0.1.0.sql ./sql/pgxntool-test--0.1.0--0.1.1.sql ./pgxntool-test.control  '@PG_LOCATION@/share/extension/'
INVOCATION OF pg_regress
(using postmaster on Unix socket, default port)
============== dropping database "contrib_regression" ==============
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== installing plpgsql                     ==============
CREATE LANGUAGE
============== running regression test queries        ==============
test pgxntool-test            ... FAILED

======================
 1 of 1 tests failed. 
======================

The differences that caused some tests to fail can be viewed in the
file "@TEST_DIR@/repo/test/regression.diffs".  A copy of the test summary that you see
above is saved in the file "@TEST_DIR@/repo/test/regression.out".

make[1]: [installcheck] Error 1 (ignored)
*** @TEST_DIR@/repo/test/expected/pgxntool-test.out
--- @TEST_DIR@/repo/test/results/pgxntool-test.out
***************
*** 57,60 ****
  \echo # TRANSACTION INTENTIONALLY LEFT OPEN!
  # TRANSACTION INTENTIONALLY LEFT OPEN!
  -- vi: expandtab ts=2 sw=2
- 
--- 57,59 ----

======================================================================

######################################
# ^^^ Should have a diff ^^^
######################################
rm -rf  sql/pgxntool-test--0.1.0.sql
rm -rf results/ regression.diffs regression.out tmp_check/ log/
cp sql/pgxntool-test.sql sql/pgxntool-test--0.1.0.sql
@INSTALL@ -c -d '@PG_LOCATION@/share/extension'
@INSTALL@ -c -m 644 ./sql/pgxntool-test--0.1.0.sql ./sql/pgxntool-test--0.1.0--0.1.1.sql ./pgxntool-test.control  '@PG_LOCATION@/share/extension/'
INVOCATION OF pg_regress
(using postmaster on Unix socket, default port)
============== dropping database "contrib_regression" ==============
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== installing plpgsql                     ==============
CREATE LANGUAGE
============== running regression test queries        ==============
test pgxntool-test            ... FAILED

======================
 1 of 1 tests failed. 
======================

The differences that caused some tests to fail can be viewed in the
file "@TEST_DIR@/repo/test/regression.diffs".  A copy of the test summary that you see
above is saved in the file "@TEST_DIR@/repo/test/regression.out".

make[1]: [installcheck] Error 1 (ignored)
*** @TEST_DIR@/repo/test/expected/pgxntool-test.out
--- @TEST_DIR@/repo/test/results/pgxntool-test.out
***************
*** 57,60 ****
  \echo # TRANSACTION INTENTIONALLY LEFT OPEN!
  # TRANSACTION INTENTIONALLY LEFT OPEN!
  -- vi: expandtab ts=2 sw=2
- 
--- 57,59 ----

======================================================================

rsync -rlpgovP test/results/ test/expected
sending incremental file list
pgxntool-test.out
RSYNC OUTPUT

sent 1,851 bytes  received 35 bytes  3,772.00 bytes/sec
total size is 1,728  speedup is 0.92
rm -rf  sql/pgxntool-test--0.1.0.sql
rm -rf results/ regression.diffs regression.out tmp_check/ log/
cp sql/pgxntool-test.sql sql/pgxntool-test--0.1.0.sql
@INSTALL@ -c -d '@PG_LOCATION@/share/extension'
@INSTALL@ -c -m 644 ./sql/pgxntool-test--0.1.0.sql ./sql/pgxntool-test--0.1.0--0.1.1.sql ./pgxntool-test.control  '@PG_LOCATION@/share/extension/'
INVOCATION OF pg_regress
(using postmaster on Unix socket, default port)
============== dropping database "contrib_regression" ==============
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== installing plpgsql                     ==============
CREATE LANGUAGE
============== running regression test queries        ==============
test pgxntool-test            ... ok

=====================
 All 1 tests passed. 
=====================

######################################
# ^^^ Should be clean output, BUT NOTE THERE WILL BE A FAILURE DIRECTLY ABOVE! ^^^
######################################
